lib/uefi/switch_stack.S

```diff
@@ -0,0 +1,23 @@
+#include <lk/asm.h>
+
+
+// int call_with_stack(void *stack, int (*fp)(), int param1, int param2);
+FUNCTION(call_with_stack)
+sub     sp,sp,16
```

Comment:
no need to use a separate sub instruction here.

can replace with 
stp    fp, lp, [sp, #-16]!